pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/TorchToMhlo.cpp

```diff
@@ -854,6 +893,65 @@ LogicalResult ConvertAtenOp<AtenDropoutOp>::matchAndRewrite(
   return success();
 }
 
+// Ref: https://pytorch.org/docs/stable/generated/torch.Tensor.expand.html
+// aten.broadcast_to has similar semantics with torch.expand
+template <>
+LogicalResult ConvertAtenOp<AtenBroadcastToOp>::matchAndRewrite(
+    AtenBroadcastToOp op,
+    OpAdaptor adaptor,
+    ConversionPatternRewriter& rewriter) const {
+  // Not a tensor type.
+  //
+  auto selfType = adaptor.self().getType().dyn_cast<TensorType>();
+  if (!selfType)
+    return op.emitError("Only tensor types are currently supported");
+
+  SmallVector<Value> dimsSize;
+  if (!getListConstructElements(adaptor.size(), dimsSize)) {
+    return op.emitError("Dims size must be a list of Scalar");
+  }
+
+  auto loc = op.getLoc();
+  auto rankType = selfType.dyn_cast<RankedTensorType>();
+  auto self_rank = rankType ? rankType.getRank() : 0;
+  auto new_rank = dimsSize.size();
+  auto leading_rank = new_rank - self_rank;
+  for (size_t d = 0; d < new_rank; ++d) {
+    auto dsize = dimsSize[d];
+    int64_t dval;
+    if (matchPattern(dsize, m_TorchConstantInt(&dval)) && dval == -1) {
+      if (d < leading_rank) {
+        return op.emitError(
+            "For the new leading dimensions, the size cannot be set to -1.");
+      } else {
+        // Passing -1 as the size for a dimension means not changing the size
+        // of that dimension.
+        dsize = rewriter.create<tensor::DimOp>(
+            loc, adaptor.self(), d - leading_rank);
+      }
+    } else {
+      dsize = rewriter.create<ToI64Op>(loc, dsize).getResult();
```

Comment:
no need to consider `leading_rank` when it's not constant?

Reply:
If the leading dimension `dsize` to expand is not a -1, we will use the `dsize` directly.